coralproject/talk
# [next] Email

## What does this PR do?

- Adds support for emails that are localized with a potential of overriding the HTML templates per tenant via a custom [nunjucks loader](https://mozilla.github.io/nunjucks/api.html#loader).
  - Reset Password
  - Confirm Email
  - Ban
  - Suspend

- Add support for `Forgot Email` on the client
  - Add support for TalkError handling for REST calls
  - Add new `account` bundle

## How do I test this PR?

### Prerequisite
Update your tenant model in the database with the following email settings:
```json
  "email": {
    "enabled": true,
    "smtpURI": "smtp://localhost:2500",
    "fromAddress": "community@test.com"
  },
```
The above settings is configured to access the inbucket server that you can run using the following docker command

```sh
#!/bin/sh
docker run -d  -p 9000:9000 -p 2500:2500 -p 1100:1100  --name inbucket inbucket/inbucket 
```

### Test feature
- Go to comment stream and in the auth popup kickoff the forgot password process.
- Check `inbucket` at `http://localhost:9000` for the received email to click on the link supplied in the email.
- Change your password
